pkg/analyzer/detection/detection.go

```diff
@@ -25,40 +26,63 @@ func DetectComponents(name string, files analyzer.Files, parent *database.Layer,
 	namespace := DetectNamespace(name, files, parent, uncertifiedRHEL)
 
 	var featureVersions []database.FeatureVersion
-	var rhelfeatures *database.RHELv2Components
+	var rhelFeatures *database.RHELv2Components
+	var err error
 
 	// In the current state, RHCOS will always be handled as certified system.
 	// If no CPEs are found on RHCOS, a note needs to be added that informs users of it.
 	// Also, the bool logic can be refactored into a single UseCertifiedWorkflow function.
 	// TODO(ROX-13906, ROX-14028): Implement note and refactor
-	if namespace != nil && (wellknownnamespaces.IsRHELNamespace(namespace.Name) || wellknownnamespaces.IsRHCOSNamespace(namespace.Name)) {
-		// This is a certified image that needs to be scanned differently.
-		// Use the RHELv2 scanner instead.
-		packages, cpes, err := rpm.ListFeatures(files)
+	if isCertifiedRHELNamespace(namespace) {
+		rhelFeatures, languageComponents, err = detectAndAnnotateCertifiedRHELComponents(name, files, namespace, languageComponents)
 		if err != nil {
 			return nil, nil, nil, nil, err
 		}
-		rhelfeatures = &database.RHELv2Components{
-			Dist:     namespace.Name,
-			Packages: packages,
-			CPEs:     cpes,
-		}
-		logrus.WithFields(logrus.Fields{LogLayerName: name, "rhel package count": len(packages), "rhel cpe count": len(cpes)}).Debug("detected rhelv2 features")
-		if err := rpm.AnnotateComponentsWithPackageManagerInfo(files, languageComponents); err != nil {
-			logrus.WithError(err).Errorf("Failed to analyze package manager info for language components: %s", name)
-		}
+		logrus.WithFields(logrus.Fields{
+			LogLayerName:         name,
+			"rhel package count": len(rhelFeatures.Packages),
+			"rhel cpe count":     len(rhelFeatures.CPEs),
+			"rhel content sets":  len(rhelFeatures.ContentSets),
+		}).Debug("detected rhelv2 features")
 	} else {
-		var err error
-		// Detect features.
 		featureVersions, err = detectFeatureVersions(name, files, namespace, parent)
 		if err != nil {
 			return nil, nil, nil, nil, err
 		}
-		if len(featureVersions) > 0 {
-			logrus.WithFields(logrus.Fields{LogLayerName: name, "feature count": len(featureVersions)}).Debug("detected features")
-		}
+		logrus.WithFields(logrus.Fields{
+			LogLayerName:    name,
```

Comment:
I'm actually not familiar with this key

Reply:
Do you mean we should change it or remove it? Not clear what your suggestion is.